crates/driver/src/db.rs

```diff
@@ -87,6 +147,95 @@ impl DriverDataBase {
         (ingot, input_file)
     }
 
+    pub fn local_ingot(
+        &mut self,
+        path: &Utf8Path,
+        version: &Version,
+        source_root: &Utf8Path,
+        source_files: Vec<(Utf8PathBuf, String)>,
+        core_ingot: InputIngot,
+    ) -> (InputIngot, IndexSet<InputFile>) {
+        let core_dependency = IngotDependency::new("core", core_ingot);
+        let mut external_ingots = IndexSet::default();
+        external_ingots.insert(core_dependency);
+        let input_ingot = InputIngot::new(
+            self,
+            path.as_str(),
+            IngotKind::Local,
+            version.clone(),
+            external_ingots,
+        );
+
+        let input_files = self.set_ingot_source_files(input_ingot, source_root, source_files);
+        (input_ingot, input_files)
+    }
+
+    pub fn core_ingot(
+        &mut self,
+        path: &Utf8Path,
+        version: &Version,
+        source_root: &Utf8Path,
+        source_files: Vec<(Utf8PathBuf, String)>,
+    ) -> (InputIngot, IndexSet<InputFile>) {
+        let input_ingot = InputIngot::new(
+            self,
+            path.as_str(),
+            IngotKind::Core,
+            version.clone(),
+            IndexSet::default(),
+        );
+
+        let input_files = self.set_ingot_source_files(input_ingot, source_root, source_files);
+        (input_ingot, input_files)
+    }
+
+    pub fn static_core_ingot(&mut self) -> (InputIngot, IndexSet<InputFile>) {
+        let core = CORE.dirs().collect::<Vec<_>>()[0];
```

Comment:
if something fails here it should panic and instruct the user to specify the core path on the cli to view resolution errors